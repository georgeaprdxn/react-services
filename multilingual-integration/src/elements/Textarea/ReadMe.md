Textarea

```jsx inside Markdown
const [val, setVal] = React.useState('Some text')
const changeHandler = e => setVal(e.target.value)
;<Textarea row={3} val={val} handleChange={e => changeHandler(e)} />
```
