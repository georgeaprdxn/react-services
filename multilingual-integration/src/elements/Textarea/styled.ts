import styled from 'styled-components'

const StyledTextarea = styled.textarea``

export { StyledTextarea }
