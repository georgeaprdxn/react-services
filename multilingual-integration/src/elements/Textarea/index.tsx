import React, { FunctionComponent } from 'react'
import { StyledTextarea } from './styled'

type Props = {
	/**
	 * number of rows
	 */
	rows?: number
	/**
	 * textarea value
	 */
	val: string
	/**
	 * handles change event
	 */
	handleChange?: (e: React.ChangeEvent<HTMLTextAreaElement>) => void
}

const Textarea: FunctionComponent<Props> = ({ rows, val, handleChange }) => {
	return (
		<StyledTextarea
			rows={rows}
			value={val}
			onChange={handleChange}
			className="custom-textarea"
		/>
	)
}

Textarea.defaultProps = {
	rows: 4,
	handleChange: () => undefined,
}

export default Textarea
