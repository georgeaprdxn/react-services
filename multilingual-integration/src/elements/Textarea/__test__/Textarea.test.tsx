import React from 'react'
import { shallow } from 'enzyme'

import Textarea from '../index'
import mockTextareaData from './Textarea.mock.json'

describe('Textarea tests', () => {
	let textareaElem: any = null
	const handleChangeStub = jest.fn()
	beforeEach(() => {
		textareaElem = shallow(
			<Textarea
				rows={mockTextareaData.row}
				val={mockTextareaData.val}
				handleChange={handleChangeStub}
			/>
		)
	})

	it('should render a <textarea />', () => {
		expect(textareaElem.find('.custom-textarea').length).toEqual(1)
	})

	it('validates all props', () => {
		expect(textareaElem.find('.custom-textarea').prop('rows')).toEqual(
			mockTextareaData.row
		)
		expect(textareaElem.find('.custom-textarea').prop('value')).toEqual(
			mockTextareaData.val
		)
	})

	it('checks the DOM events', () => {
		textareaElem
			.find(`.custom-textarea`)
			.simulate('change', { target: { value: mockTextareaData.val } })

		expect(handleChangeStub).toHaveBeenCalledTimes(1)
	})

	it('matches the snaphot', () => {
		expect(textareaElem).toMatchSnapshot()
	})
})
