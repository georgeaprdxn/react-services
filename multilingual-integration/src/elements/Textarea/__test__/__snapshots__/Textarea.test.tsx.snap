// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea tests matches the snaphot 1`] = `
<styled.textarea
  className="custom-textarea"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "target": Object {
              "value": "Some Text",
            },
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  rows={4}
  value="Some Text"
/>
`;
